File: JavaScript/resistor_color_trio.js
const COLORS = ["black","brown","red","orange","yellow","green","blue","violet","grey","white"];

class ResistorColorTrio {
    constructor(colors) {
        this.colors = colors.map(col => col.toLowerCase());
    }
  
    get label() {
        let indices = [COLORS.indexOf(this.colors[0]), 
                       COLORS.indexOf(this.colors[1]), 
                       COLORS.indexOf(this.colors[2])];
        
        if(indices.includes(-1)) {
            throw new Error("invalid color");
        }

        // 'Pretify' the label
        let numString = String(indices[0]) + String(indices[1]) + '0'.repeat(indices[2]);

        if(numString.slice(-3) === '000') {
            numString = numString.replace('000', '') + ' kiloohms';
        }
        else {
            numString = numString + ' ohms';
        }
        
        return 'Resistor value: ' + numString;  
    }
}

resistors = [["orange", "orange", "black"],
             ['Yellow', 'violet', 'yellow'],
            //  ['yellow', 'purple', 'black'], // error test
            ['red', 'black', 'red'],
]

resistors.forEach(r => {
    rct = new ResistorColorTrio(r);
    console.log(rct, rct.label);
});

File: C#/Resistor Color Trio/ResistorColorTrio.cs
using System;
using System.Linq;

public static class ResistorColorTrio
{
    public static readonly string[] Colors = ["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"];

    public static string Label(string[] colors)
    {
        colors = colors.Select(c => c.ToLower()).ToArray();

        var indices = colors.Select(c => Array.IndexOf(Colors, c)).ToArray();

        // 'Pretify' the label
        int n = Convert.ToInt32(indices[0].ToString() + indices[1].ToString() + string.Concat(Enumerable.Repeat('0', indices[2])));
        string numString = "";

        if (n >= 1e9)
        {
            numString = $"{(n / 1e9).ToString("0")} gigaohms";
        }
        else if (n >= 1e6)
        {
            numString = $"{(n / 1e6).ToString("0")} megaohms";
        }
        else if (n >= 1e3)
        {
            numString = $"{(n / 1e3).ToString("0")} kiloohms";
        }
        else
        {
            numString = $"{n} ohms";
        }

        return numString;
    }
}